mico-core/src/main/java/io/github/ust/mico/core/broker/MicoApplicationBroker.java

```diff
@@ -249,17 +294,86 @@ public MicoApplication removeMicoServiceFromMicoApplicationByShortNameAndVersion
+        throws MicoApplicationNotFoundException, MicoApplicationIsNotUndeployedException, KafkaFaasConnectorVersionNotFoundException {
+
+        // Retrieve application and service from database (checks whether they exist)
+        MicoApplication micoApplication = getMicoApplicationByShortNameAndVersion(applicationShortName, applicationVersion);
+
+        // Check whether application is currently undeployed, if not it is not allowed to add a KafkaFaasConnector instance to the application
+        if (!micoKubernetesClient.isApplicationUndeployed(micoApplication)) {
+            throw new MicoApplicationIsNotUndeployedException(applicationShortName, applicationShortName);
+        }
+
+        MicoService kfConnector;
+        try {
+            kfConnector = micoServiceBroker.getServiceFromDatabase(KAFKA_FAAS_CONNECTOR_SERVICE_NAME, kfConnectorVersion);
+        } catch (MicoServiceNotFoundException e) {
+            throw new KafkaFaasConnectorVersionNotFoundException(kfConnectorVersion);
+        }
+
+        String instanceId = UIDUtils.uidFor(kfConnector);
+        MicoServiceDeploymentInfo sdi = new MicoServiceDeploymentInfo().setService(kfConnector);
+        KFConnectorDeploymentInfo kfConnectorDeploymentInfo = new KFConnectorDeploymentInfo()
+            .setServiceDeploymentInfo(sdi)
+            .setApplication(micoApplication)
+            .setInstanceId(instanceId);
+        micoApplication.getKafkaFaasConnectorDeploymentInfos().add(kfConnectorDeploymentInfo);
+        applicationRepository.save(micoApplication);
+
+        log.debug("Added KafkaFaasConnector in version '{}' with instance ID '{}' to MicoApplication '{}' '{}'." +
+                "This application now uses {} KF connector instances.",
+            kfConnectorVersion, instanceId, applicationShortName, applicationVersion,
+            micoApplication.getKafkaFaasConnectorDeploymentInfos().size());
+
+        // TODO: Set default deployment information (covered in epic mico#750)
+        // Set default deployment information (environment variables, topics)
+        //serviceDeploymentInfoBroker.setDefaultDeploymentInformationForKafkaEnabledService(sdi);
+        //serviceDeploymentInfoBroker.updateKafkaFaasConnectorDeploymentInformation(applicationShortName, applicationVersion, instanceId,
+        //    new KFConnectorDeploymentInfoRequestDTO(sdi));
+        return kfConnectorDeploymentInfo;
+    }
+
+    public void removeKafkaFaasConnectorInstanceFromMicoApplicationByInstanceId(
+        String applicationShortName, String applicationVersion, String instanceId)
+        throws MicoApplicationNotFoundException, MicoApplicationIsNotUndeployedException, KafkaFaasConnectorInstanceNotFoundException, MicoApplicationDoesNotIncludeKFConnectorInstanceException {
+
+        // Retrieve application from database (checks whether it exists)
+        MicoApplication micoApplication = checkForKfConnectorInMicoApplication(applicationShortName, applicationVersion, instanceId);
```

Comment:
I would assume a method that  starts with `check` returns a `boolean`. Maybe rename to `get`. The checking is then only mentioned in javadoc the comment above and the throws statement which should still be enough.